lib/mocha.js

```diff
@@ -1219,22 +1221,40 @@ Mocha.prototype.runGlobalTeardown = async function runGlobalTeardown(
 };
 
 /**
- * Run global fixtures sequentially with context `context`
+ * Run global fixtures sequentially with context `context`.
  * @private
  * @param {MochaGlobalFixture[]} [fixtureFns] - Fixtures to run
  * @param {object} [context] - context object
  * @returns {Promise<object>} context object
  */
 Mocha.prototype._runGlobalFixtures = async function _runGlobalFixtures(
   fixtureFns = [],
-  context = {}
+  context = {},
 ) {
-  for await (const fixtureFn of fixtureFns) {
-    await fixtureFn.call(context);
+  for (const fixtureFn of fixtureFns) {
+    try {
+      await fixtureFn.call(context);
+    } catch (err) {
+      if(context.stats) {
+        context.stats.failures++;
+        context.failures++;
+       
+        const isSetup = this.options.globalSetup && this.options.globalSetup.includes(fixtureFn);
```

Comment:
[Refactor] Trying to query `this` properties to determine the phase is a little awkward: it's several lines of code and relies on other implementation details.

`_runGlobalFixtures` is called by places that know what phase they're in: `runGlobalTeardown` and `runGlobalSetup`. How about having them pass in the phase as an argument?